### PR TITLE
Consume minSend amounts on frontend

### PR DESCRIPTION
**Pull request by davidlaprade**
Consume the minSend amounts that are sent by the backend and use them for frontend validation. Values from the backend are rounded so as to not be weirdly specific for the user.

frontend-min-send-amounts

### CODE REVIEW

File: frontend/src/pages/AccountSend.vue
Comment (davidlaprade) [inline frontend/src/pages/AccountSend.vue:318]: This number should probably be revised. The live values are at least 5x as high. In the event that the backend went down and we didn't have live minSends beind sent, we'd probably still want to enforce the high estimates

File: frontend/test/utils.test.ts
Comment (davidlaprade) [inline frontend/test/utils.test.ts:160]: I'm curious if anyone disagrees with these expectations RE how we round minSend amounts for users

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:100]: Is there a chance we ever add more data to these types? If so, maybe we rename to `TokenInfoExtended` and `TokenListExtended` or something similar

File: frontend/src/components/models.ts
Comment (mds1) [inline frontend/src/components/models.ts:105]: Should the backend just return the native token info and min send amount just like any other token, instead of special casing it? Right now in `wallet.ts` we manually extend the array of tokens to add the native token, but that doesn't seem necessary/ideal

File: frontend/src/pages/AccountSend.vue
Comment (mds1) [inline frontend/src/pages/AccountSend.vue:302]: I think we should get rid of this method and just key off the native token address (`0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE`) in `getTokenMinSendAmountFallback`. As you can probably tell, I'm not a fan of special-casing the native token if we don't have to since it just makes things clunkier and harder to read/maintain

File: frontend/src/pages/AccountSend.vue
Comment (mds1) [reply frontend/src/pages/AccountSend.vue:318]: I'll defer to you and @apbendi on good fallback values here

File: frontend/src/pages/AccountSend.vue
Comment (mds1) [inline frontend/src/pages/AccountSend.vue:323]: We use `chainId` enough that we should probably just add a `chainId` export from `wallet.ts` and start using that. We can do it as a computed property from `network.value.chainId`. I know this line was unmodified, so we can pull this out to a separate issue if you want

File: frontend/src/pages/AccountSend.vue
Comment (mds1) [inline frontend/src/pages/AccountSend.vue:322]: Since the relayer now includes a min send amount on the returned tokens array, this method overly complicated. It seems a cleaner approach would be:
- Update `getMinSendAmount` to take the new `TokenInfoExtended` (or whatever the name is) type as input
- Refactor `wallet.ts` so the `tokens` computed property we export is of that type, and includes min send amounts
- If the relayer is down, that `tokens` export should only have ETH with the default value we define in the frontend. Otherwise, the API is working so we have all tokens and their default min send amounts, therefore we don't need to define token overrides anywhere on the frontend aside from native token overrides in `wallet.ts` (the assumption here is that we'll never get back an array of tokens without min send amounts from the API—if we currently can, then we should move the defining of token minimums to the backend so that doesn't happen, and just duplicate the native token min amount on the frontend)
- Update this method to just parse the `tokens` array for the minimum send amount and humanize it

File: frontend/src/utils/relayer.ts
Comment (mds1) [inline frontend/src/utils/relayer.ts:17]: Since we also use OZ now, this is no longer ITX specific so we should probably rename this class this to `UmbraRelayer` or or similar

File: frontend/src/utils/relayer.ts
Comment (mds1) [inline frontend/src/utils/relayer.ts:22]: Same comment about not needing to special-case the native token and just include it in the tokens array

**Review by mds1 — COMMENTED** [review]
Also just want to note that I'd prefer if we can merge #293 before this one, just in case there's merge conflicts or anything since I think #293 is more "sensitive"

File: frontend/src/components/models.ts
Comment (davidlaprade) [reply frontend/src/components/models.ts:100]: Good call, I agree

File: frontend/src/store/wallet.ts
Comment (mds1) [inline frontend/src/store/wallet.ts:91]: `newChainId` seems like a more descriptive name

File: frontend/src/store/wallet.ts
Comment (mds1) [inline frontend/src/store/wallet.ts:385]: How do these compare to the prior defaults / how did we come up with these?

File: frontend/src/store/wallet.ts
Comment (mds1) [inline frontend/src/store/wallet.ts:403]: Can we make sure the native token is always first in this list? Since I believe this order is the order used in the token dropdown. I think native token first, sort alphabetically by symbol for the rest would be great.

File: frontend/src/utils/umbra_api.ts
Comment (mds1) [inline frontend/src/utils/umbra_api.ts:40]: same question about special cased native token min send amount

File: frontend/src/components/models.ts
Comment (davidlaprade) [reply frontend/src/components/models.ts:136]: We have to because that's what the backend is currently returning. Until the backend is updated (which I plan to do later) this is what we need to use

File: frontend/src/store/wallet.ts
Comment (davidlaprade) [reply frontend/src/store/wallet.ts:385]: These are the current values returned by the backend, calculated using the formula:

Gas Price x Estimated Gas Limit x Scale Factor

I'll add a note about it

File: frontend/src/components/AccountReceiveTable.vue
Comment (davidlaprade) [reply frontend/src/components/AccountReceiveTable.vue:557]: I'm not honestly sure

File: frontend/src/pages/AccountSend.vue
Comment (davidlaprade) [reply frontend/src/pages/AccountSend.vue:325]: I agree. I had initially taken that approach, but then that led to noisy errors popping up when switching chains (I believe) because there was a race condition between the provider returning chain data and the backend returning token data. That might not be a problem anymore though. Let me double check...

File: frontend/src/pages/AccountSend.vue
Comment (davidlaprade) [reply frontend/src/pages/AccountSend.vue:264]: I had needed this to avoid some noisy errors, but when I just removed it now I wasn't able to reproduce them. Not sure what happened exactly. I'll leave it out

File: frontend/src/components/models.ts
Comment (mds1) [reply frontend/src/components/models.ts:136]: Are we tracking this in an issue somewhere, either here or in the backend?

File: frontend/src/components/AccountReceiveTable.vue
Comment (mds1) [reply frontend/src/components/AccountReceiveTable.vue:557]: @apbendi Can you share the mapping between network => relayer/tx identifier used? Once we have that we should pull this out into a separate issue so we don't lose track

Comment (apbendi) [reply frontend/src/components/AccountReceiveTable.vue:557]: With ITX (on Mainnet/Rinkeby), their API returns a custom hash and you have to subsequently query their endpoint to figure out the list of actual network transaction hashes. That is what we pass forward from Umbra API.

With Defender, we get back the hash of the first transaction submitted, and that is what we pass forward from Umbra API. Defender [also](https://docs.openzeppelin.com/defender/relay#transaction-ids) gives you a separate, custom Identifier which can use to query their API, to find out if that first transaction has been replaced or cancelled. We currently don't expose this in any response from Umbra API.

In practice, since we're using this on Polygon and using `fast`, this virtually never happens, and so the hash we get back ends up being the hash of the transaction that is ultimately mined. This should also be the case on Optimism and Arbitrum.

If we ever started using Defender on mainnet, we might have to revisit this assumptino.

File: frontend/src/pages/AccountSend.vue
Comment (apbendi) [reply frontend/src/pages/AccountSend.vue]: Sorry this diff is outdated now and I'm not clear what was being asked. Is my input still needed on something here?

Comment (mds1) [reply frontend/src/pages/AccountSend.vue]: Umm I'm not certain either haha, deferring to @davidlaprade 😅

File: frontend/src/components/models.ts
Comment (apbendi) [reply frontend/src/components/models.ts:136]: Just created [this issue](https://github.com/ScopeLift/umbra-api/issues/51) on the API side to track it

Comment (davidlaprade) [reply frontend/src/components/models.ts:136]: Looks great, thanks @apbendi

File: frontend/src/pages/AccountSend.vue
Comment (davidlaprade) [reply frontend/src/pages/AccountSend.vue:325]: It's no longer a problem. Removed 👍

File: frontend/src/components/AccountReceiveTable.vue
Comment (mds1) [reply frontend/src/components/AccountReceiveTable.vue:557]: Created https://github.com/ScopeLift/umbra-protocol/issues/329 to track this and make sure the logic in this code block is correct

File: frontend/test/utils.test.ts
Comment (mds1) [reply frontend/test/utils.test.ts]: This seems fine to me! @apbendi any thoughts?

File: frontend/src/utils/relayer.ts
Comment (mds1) [reply frontend/src/utils/relayer.ts]: @davidlaprade looks like this was removed, any objections to resolving this comment?

File: frontend/test/utils.test.ts
Comment (mds1) [reply frontend/test/utils.test.ts]: Actually right now on Optimism it says the minimum send is `0.0097 OETH`, but that should probably be rounded to `0.01 OETH`?

Comment (mds1) [reply frontend/test/utils.test.ts]: Though maybe not, If it was 0.0051 OETH we wouldn't want to round that up. In general, I guess it just feels like the variable min-send UX is kinda confusing. Gas prices might be at a low when I send funds, then they're basically stuck. It seems better to have a consistent min send based on e.g. the 90th percentile gas price, so the floor is constant and 90% of the time we know users will be able to withdraw. And then we just periodically update that.

But, this can be discussed in a separate issue

Comment (mds1) [reply frontend/test/utils.test.ts]: Ahh got it, sounds good. Yes I'll resolve now!

**Review by apbendi — APPROVED** [review]
Working well. Thanks @davidlaprade 👍

File: frontend/src/pages/AccountSend.vue
Comment (davidlaprade) [reply frontend/src/pages/AccountSend.vue]: I revised the numbers so they were close to recent minSend values from the backend. No further action needed. Thanks, guys!

**Review by mds1 — APPROVED** [review]

**Pull request merged**
